In [1]:
import pandas as pd
import numpy as np

In [2]:
df = pd.read_excel("/home/jupyter/src/dataviz/podcastcount/2LMarketBasket.xlsx")

In [3]:
df.shape

(10931, 44)

In [4]:
df.rename(columns = {'2L+ TKI Target':'2Lplus',
                     'LTD Reach - Live':'LiveRepTouch',
                     'LTD Reach - All Reach Methods':'AnyTouch',
                     'LTD QINLOCK Rx (HCP)' : 'WroteQinlock',
                     'April Marketing Campaign - Email Open Count':'AprilEmailOpens'},inplace = True)

In [5]:
list(df)

['NPI',
 'HCP ID',
 'First Name',
 'Last Name',
 'Specialty',
 'Sales Tier',
 'Facility Tier',
 '2Lplus',
 'AnyTouch',
 'LiveRepTouch',
 'AprilEmailOpens',
 'Total GIST Patients',
 'Total Treated GIST Patients',
 'Total 3L+ GIST Patients',
 'Total Ayva GIST Patients',
 'Komodo Clinical Experience Score',
 'Komodo Clinical Volume Score',
 'GLEEVEC Units - 52 Week',
 'GLEEVEC Units - 26 Week',
 'Sutent Units - 52 Week',
 'Sutent Units - 26 Week',
 'Stivarga Units - 52 Week',
 'Stivarga Units - 26 Week',
 'Ayva Units - 52 Week',
 'Ayva Units - 26 Week',
 'Other TKI Units - 52 Week',
 'Other TKI Units - 26 Week',
 '2L+ TKI Units - 52 Week',
 '2L+ TKI Units - 26 Week',
 'Field Reported 2L+',
 'WroteQinlock',
 'Region',
 'Territory',
 'Facility ID',
 'Facility Name',
 'Facility Address',
 'Facility Address 2',
 'Facility City',
 'Facility State',
 'Facility Zip',
 'Facility Phone',
 'Facility Fax',
 'Network/Parent',
 'Facility Segment']

In [6]:
pd.pivot_table(df,index=["2Lplus","Region"])

2L+ TKI Units - 26 Week  2L+ TKI Units - 52 Week  \
2Lplus Region                                                          
No     Central                     0.132107                 0.244147   
       Great Lakes                 0.209841                 0.399421   
       Northeast                   0.023739                 0.051929   
       South                       0.161572                 0.269287   
       Southeast                   0.082982                 0.205345   
       West                        0.219638                 0.400517   
Yes    Central                     2.041885                 3.921466   
       Great Lakes                 2.004902                 4.004902   
       Northeast                   2.657754                 4.791444   
       South                       1.932990                 3.886598   
       Southeast                   1.862559                 3.530806   
       West                        1.322581                 2.562724   

                    AnyTouch  AprilEmailOpens  Ayva Units - 26 Week  \
2Lplus Region                                                         
No     Central      1.774658         3.540000              0.005017   
       Great Lakes  2.514036         3.204724              0.000000   
       Northeast    1.439294         3.120846              0.000000   
       South        2.276278         3.047210              0.000000   
       Southeast    3.586261         3.040590              0.000000   
       West         1.839329         3.388889              0.002584   
Yes    Central      5.138393         3.181818              0.026178   
       Great Lakes  6.476596         3.638298              0.014706   
       Northeast    4.676190         2.808511              0.000000   
       South        6.131455         3.409091              0.041237   
       Southeast    8.902834         2.756098              0.052133   
       West         4.927978         3.016129              0.050179   

                    Ayva Units - 52 Week  Facility Zip  Field Reported 2L+  \
2Lplus Region                                                                
No     Central                  0.005017  64390.398630            0.000000   
       Great Lakes              0.000000  34963.549595            0.000000   
       Northeast                0.000000  10198.255519            0.000000   
       South                    0.000000  70357.611506            0.000000   
       Southeast                0.000000  31826.834819            0.000000   
       West                     0.002584  92779.757794            0.000000   
Yes    Central                  0.047120  60466.223214            1.489247   
       Great Lakes              0.014706  34494.000000            1.656805   
       Northeast                0.016043   9371.661905            1.333333   
       South                    0.103093  71173.211268            1.356688   
       Southeast                0.061611  31603.441296            1.241935   
       West                     0.064516  93192.650970            1.484746   

                    GLEEVEC Units - 26 Week  GLEEVEC Units - 52 Week  ...  \
2Lplus Region                                                         ...   
No     Central                     2.070234                 4.133779  ...   
       Great Lakes                 2.043415                 4.046310  ...   
       Northeast                   2.243323                 4.379822  ...   
       South                       2.314410                 4.499272  ...   
       Southeast                   2.032349                 3.893108  ...   
       West                        2.240310                 4.447028  ...   
Yes    Central                     4.062827                 8.272251  ...   
       Great Lakes                 5.171569                10.421569  ...   
       Northeast                   6.759358                13.572193  ...   
       South                       4.577320                 9.195876  ...   

In [7]:
pd.pivot_table(df,index=["2Lplus","Region"],values=["AprilEmailOpens"],aggfunc=[np.sum])

sum
                   AprilEmailOpens
2Lplus Region                     
No     Central               885.0
       Great Lakes           814.0
       Northeast            1033.0
       South                 710.0
       Southeast             824.0
       West                  671.0
Yes    Central               140.0
       Great Lakes           171.0
       Northeast             132.0
       South                 150.0
       Southeast             113.0
       West                  187.0

In [8]:
pd.pivot_table(df,index=["2Lplus","Region"],values=["Total 3L+ GIST Patients","AprilEmailOpens"],aggfunc=[np.sum])

sum                        
                   AprilEmailOpens Total 3L+ GIST Patients
2Lplus Region                                             
No     Central               885.0                     6.0
       Great Lakes           814.0                    10.0
       Northeast            1033.0                     1.0
       South                 710.0                     7.0
       Southeast             824.0                     9.0
       West                  671.0                     8.0
Yes    Central               140.0                    31.0
       Great Lakes           171.0                    44.0
       Northeast             132.0                    53.0
       South                 150.0                    45.0
       Southeast             113.0                    49.0
       West                  187.0                    34.0

In [9]:
pd.pivot_table(df,index=["2Lplus","Region"],
               values=["Total 3L+ GIST Patients","Total Treated GIST Patients"],
               aggfunc=[np.sum])

sum                            
                   Total 3L+ GIST Patients Total Treated GIST Patients
2Lplus Region                                                         
No     Central                         6.0                       418.0
       Great Lakes                    10.0                       499.0
       Northeast                       1.0                       502.0
       South                           7.0                       530.0
       Southeast                       9.0                       488.0
       West                            8.0                       327.0
Yes    Central                        31.0                       382.0
       Great Lakes                    44.0                       472.0
       Northeast                      53.0                       610.0
       South                          45.0                       431.0
       Southeast                      49.0                       524.0
       West                           34.0                       522.0

In [10]:
df["2Lplus"].value_counts()

No     9441
Yes    1490
Name: 2Lplus, dtype: int64

+ (41) 3L Patients among 9441 HCPs **THAT ARE NOT** 2L plus
+ (256) 3L Patients among 1490 HCPs **THAT ARE** 2L plus

In [11]:
pd.pivot_table(df,index=["2Lplus"],
               values=["Total 3L+ GIST Patients"],
               columns=["Region"],
               aggfunc=[np.sum])

sum                                            
       Total 3L+ GIST Patients                                            
Region                 Central Great Lakes Northeast South Southeast  West
2Lplus                                                                    
No                         6.0        10.0       1.0   7.0       9.0   8.0
Yes                       31.0        44.0      53.0  45.0      49.0  34.0

In [12]:
table = pd.pivot_table(df,index=["2Lplus","Region"],
               values=["Total 3L+ GIST Patients","Total Treated GIST Patients","LiveRepTouch","AnyTouch"],
               aggfunc=[np.sum])
table

sum                                       \
                   AnyTouch LiveRepTouch Total 3L+ GIST Patients   
2Lplus Region                                                      
No     Central         2591          652                     6.0   
       Great Lakes     4030          801                    10.0   
       Northeast       2608          234                     1.0   
       South           3205          492                     7.0   
       Southeast       6839         1159                     9.0   
       West            2301          544                     8.0   
Yes    Central         1151          408                    31.0   
       Great Lakes     1522          340                    44.0   
       Northeast        982          159                    53.0   
       South           1306          262                    45.0   
       Southeast       2199          550                    49.0   
       West            1779          477                    34.0   

                                                
                   Total Treated GIST Patients  
2Lplus Region                                   
No     Central                           418.0  
       Great Lakes                       499.0  
       Northeast                         502.0  
       South                             530.0  
       Southeast                         488.0  
       West                              327.0  
Yes    Central                           382.0  
       Great Lakes                       472.0  
       Northeast                         610.0  
       South                             431.0  
       Southeast                         524.0  
       West                              522.0

In [13]:
table = pd.pivot_table(df,index=["2Lplus","Region"],
               values=["Total 3L+ GIST Patients","LiveRepTouch","AnyTouch","AprilEmailOpens"],
               aggfunc=[np.sum])
table

sum                               \
                   AnyTouch AprilEmailOpens LiveRepTouch   
2Lplus Region                                              
No     Central         2591           885.0          652   
       Great Lakes     4030           814.0          801   
       Northeast       2608          1033.0          234   
       South           3205           710.0          492   
       Southeast       6839           824.0         1159   
       West            2301           671.0          544   
Yes    Central         1151           140.0          408   
       Great Lakes     1522           171.0          340   
       Northeast        982           132.0          159   
       South           1306           150.0          262   
       Southeast       2199           113.0          550   
       West            1779           187.0          477   

                                            
                   Total 3L+ GIST Patients  
2Lplus Region                               
No     Central                         6.0  
       Great Lakes                    10.0  
       Northeast                       1.0  
       South                           7.0  
       Southeast                       9.0  
       West                            8.0  
Yes    Central                        31.0  
       Great Lakes                    44.0  
       Northeast                      53.0  
       South                          45.0  
       Southeast                      49.0  
       West                           34.0

In [14]:
list(table)

[('sum', 'AnyTouch'),
 ('sum', 'AprilEmailOpens'),
 ('sum', 'LiveRepTouch'),
 ('sum', 'Total 3L+ GIST Patients')]

In [15]:
table

sum                               \
                   AnyTouch AprilEmailOpens LiveRepTouch   
2Lplus Region                                              
No     Central         2591           885.0          652   
       Great Lakes     4030           814.0          801   
       Northeast       2608          1033.0          234   
       South           3205           710.0          492   
       Southeast       6839           824.0         1159   
       West            2301           671.0          544   
Yes    Central         1151           140.0          408   
       Great Lakes     1522           171.0          340   
       Northeast        982           132.0          159   
       South           1306           150.0          262   
       Southeast       2199           113.0          550   
       West            1779           187.0          477   

                                            
                   Total 3L+ GIST Patients  
2Lplus Region                               
No     Central                         6.0  
       Great Lakes                    10.0  
       Northeast                       1.0  
       South                           7.0  
       Southeast                       9.0  
       West                            8.0  
Yes    Central                        31.0  
       Great Lakes                    44.0  
       Northeast                      53.0  
       South                          45.0  
       Southeast                      49.0  
       West                           34.0

In [16]:
settings = pd.pivot_table(df,index=["2Lplus","Region"],
                values=["Total 3L+ GIST Patients"],
                columns=["Facility Segment"],
                aggfunc=[len])
settings

len                                 \
                   Total 3L+ GIST Patients                                  
Facility Segment                  AMC Flag IDN Flag Independent Hospitals   
2Lplus Region                                                               
No     Central                       271.0    802.0                  31.0   
       Great Lakes                   414.0    785.0                  39.0   
       Northeast                     753.0    750.0                  55.0   
       South                         250.0    663.0                  35.0   
       Southeast                     313.0    803.0                  68.0   
       West                          265.0    548.0                  87.0   
Yes    Central                        57.0    104.0                   7.0   
       Great Lakes                    57.0    111.0                   4.0   
       Northeast                     114.0     65.0                   6.0   
       South                          44.0     98.0                   7.0   
       Southeast                      49.0    101.0                   9.0   
       West                           76.0    160.0                  23.0   

                                             
                                             
Facility Segment   Physician Group Practice  
2Lplus Region                                
No     Central                        282.0  
       Great Lakes                    216.0  
       Northeast                      210.0  
       South                          390.0  
       Southeast                      552.0  
       West                           228.0  
Yes    Central                         49.0  
       Great Lakes                     44.0  
       Northeast                       19.0  
       South                           47.0  
       Southeast                       66.0  
       West                            78.0

In [17]:
secondlineplus = df[df["2Lplus"] == "Yes"]
secondlineplus.shape

(1490, 44)

In [18]:
scripts = pd.pivot_table(secondlineplus,index=["WroteQinlock","Region"],
                values=["Total 3L+ GIST Patients"],
                columns=["Facility Segment"],                         
                aggfunc=[len])
scripts

len           \
                         Total 3L+ GIST Patients            
Facility Segment                        AMC Flag IDN Flag   
WroteQinlock Region                                         
No           Central                        34.0     77.0   
             Great Lakes                    33.0     91.0   
             Northeast                      65.0     57.0   
             South                          26.0     82.0   
             Southeast                      30.0     79.0   
             West                           59.0    129.0   
Yes          Central                        23.0     27.0   
             Great Lakes                    24.0     20.0   
             Northeast                      49.0      8.0   
             South                          18.0     16.0   
             Southeast                      19.0     22.0   
             West                           17.0     31.0   

                                                                         
                                                                         
Facility Segment         Independent Hospitals Physician Group Practice  
WroteQinlock Region                                                      
No           Central                       6.0                     40.0  
             Great Lakes                   2.0                     33.0  
             Northeast                     5.0                     13.0  
             South                         6.0                     36.0  
             Southeast                     5.0                     50.0  
             West                         19.0                     71.0  
Yes          Central                       1.0                      9.0  
             Great Lakes                   2.0                     11.0  
             Northeast                     1.0                      6.0  
             South                         1.0                     11.0  
             Southeast                     4.0                     16.0  
             West                          4.0                      7.0

In [43]:
thirdLnoQ = df[(df["2Lplus"] == "Yes")
    & (df["Total 3L+ GIST Patients"] > 0)
    & (df["WroteQinlock"]=="No")]
thirdLnoQ.shape

(132, 44)

In [44]:
thirdLsettings = pd.pivot_table(thirdLnoQ,index=["Region"],
                values=["Total 3L+ GIST Patients"],
                columns=["Facility Segment"],                         
                aggfunc=[len])
thirdLsettings

len                                 \
                 Total 3L+ GIST Patients                                  
Facility Segment                AMC Flag IDN Flag Independent Hospitals   
Region                                                                    
Central                              2.0     10.0                   NaN   
Great Lakes                          6.0     13.0                   NaN   
Northeast                           12.0     12.0                   1.0   
South                                3.0      9.0                   1.0   
Southeast                            5.0     16.0                   NaN   
West                                 4.0      5.0                   2.0   

                                           
                                           
Facility Segment Physician Group Practice  
Region                                     
Central                               2.0  
Great Lakes                           3.0  
Northeast                             1.0  
South                                 3.0  
Southeast                             6.0  
West                                 10.0

In [45]:
thirdLnoQidn = df[(df["2Lplus"] == "Yes")
    & (df["Total 3L+ GIST Patients"] > 0)
    & (df["WroteQinlock"]=="No")
    & (df["Facility Segment"]=="IDN Flag")]
thirdLnoQidn.shape

SyntaxError: invalid syntax (<ipython-input-45-48144a6b3a7c>, line 1)

In [39]:
thirdLnoQidn

NPI      HCP ID First Name Last Name              Specialty  \
72     1528234143  DA01003353       SETH   POLLACK  HEMATOLOGY & ONCOLOGY   
81     1740228857  DA01007552      SAMIR   UNDEVIA  HEMATOLOGY & ONCOLOGY   
707    1568453298  DA01010233      MALIK  BANDEALY  HEMATOLOGY & ONCOLOGY   
710    1497189468  DA01010234    ALLISON    HARDIN  HEMATOLOGY & ONCOLOGY   
711    1588655351  DA01001549      KELLY   KONESKO     NURSE PRACTITIONER   
...           ...         ...        ...       ...                    ...   
9820   1235101437  DA01000173       GARY       SHI      ONCOLOGY, MEDICAL   
10362  1508889338  DA01001859      JARED     ACOBA  HEMATOLOGY & ONCOLOGY   
10421  1508833971  DA01001158     MELVIN   PALALAY  HEMATOLOGY & ONCOLOGY   
10602  1265610752  DA01003444      JORGE    CHAVES  HEMATOLOGY & ONCOLOGY   
10610  1861587339  DA01003394      KAREN      HUNT  HEMATOLOGY & ONCOLOGY   

      Sales Tier Facility Tier 2Lplus  AnyTouch  LiveRepTouch  ...  \
72        TIER 2        TIER 1     No         4             1  ...   
81        TIER 2        TIER 2     No         6             1  ...   
707       TIER 3        TIER 3    Yes         9             2  ...   
710       TIER 3        TIER 3    Yes         8             1  ...   
711       TIER 3        TIER 2    Yes        12             4  ...   
...          ...           ...    ...       ...           ...  ...   
9820      TIER 2        TIER 2    Yes         9             3  ...   
10362     TIER 1        TIER 1    Yes         8             2  ...   
10421     TIER 1        TIER 1     No        12             4  ...   
10602     TIER 2        TIER 1    Yes        14             3  ...   
10610     TIER 2        TIER 2    Yes         8             0  ...   

                           Facility Name     Facility Address  \
72        Northwestern Memorial Hospital       251 E HURON ST   
81                       Edward Hospital  801 S WASHINGTON ST   
707         Aurora Cancer Care - Kenosha       6811 118TH AVE   
710         Aurora Cancer Care - Kenosha       6811 118TH AVE   
711          Aurora Cancer Care - Racine  8400 WASHINGTON AVE   
...                                  ...                  ...   
9820              Sutter Amador Hospital     200 MISSION BLVD   
10362              Queens Medical Center    1301 PUNCHBOWL ST   
10421              Queens Medical Center    1301 PUNCHBOWL ST   
10602           St Joseph Medical Center          1717 S J ST   
10610  Kalispell Regional Medical Center     310 SUNNYVIEW LN   

       Facility Address 2   Facility City  Facility State  Facility Zip  \
72                    NaN         CHICAGO              IL         60611   
81                    NaN      NAPERVILLE              IL         60540   
707                   NaN         KENOSHA              WI         53142   
710                   NaN         KENOSHA              WI         53142   
711                   NaN  MOUNT PLEASANT              WI         53406   
...                   ...             ...             ...           ...   
9820                  NaN         JACKSON              CA         95642   
10362                 NaN        HONOLULU              HI         96813   
10421                 NaN        HONOLULU              HI         96813   
10602                 NaN          TACOMA              WA         98405   
10610                 NaN       KALISPELL              MT         59901   

       Facility Phone    Facility Fax                 Network/Parent  \
72     (312) 926-2520  (312) 926-6374                            NaN   
81     (630) 355-0450  (630) 868-2240                            NaN   
707    (262) 857-5700  (262) 857-1142         AURORA HEALTH CARE INC   
710    (262) 857-5700  (262) 857-1142         AURORA HEALTH CARE INC   
711    (262) 884-8340  (262) 884-8341                            NaN   
...               ...             ...                            ...   
9820   (209) 223-7500  (209) 223-7509                  SUTTER 

In [35]:
idnOpportunity = pd.pivot_table(thirdLnoQidn,index=["Network/Parent"],
                values=["Total 3L+ GIST Patients"],
                columns=["Region"],                        
                aggfunc=[len],fill_value=0)
idnOpportunity
#idnOpportunity.to_csv("/home/jupyter/src/dataviz/podcastcount/idnOpportunity.csv")

len  \
                                                   Total 3L+ GIST Patients   
Region                                                             Central   
Network/Parent                                                               
"QUEEN'S HEALTH SYSTEMS"                                                 0   
"SAINT JOSEPH'S CANDLER HEALTH SYSTEM"                                   0   
ALBERT EINSTEIN HEALTHCARE NETWORK                                       0   
ALLEGHENY HEALTH NETWORK                                                 0   
ASCENSION HEALTH                                                         0   
AURORA HEALTH CARE INC                                                   2   
BETH ISRAEL HEALTHCARE                                                   0   
CANCER TREATMENT CENTERS OF AMERICA                                      0   
CARLE FOUNDATION                                                         2   
CAROLINAS HEALTHCARE SYSTEM                                              0   
CATHOLIC HEALTH INITIATIVES                                              0   
CATHOLIC HEALTH SERVICES OF LONG ISLAND                                  0   
COMMUNITY HEALTH SYSTEMS INC                                             0   
COOPER HEALTH SYSTEM                                                     0   
COXHEALTH                                                                1   
DCH HEALTH SYSTEM                                                        0   
DIGNITY HEALTH                                                           0   
HCA HOLDINGS INC                                                         0   
HOSPITAL SISTERS HEALTH SYSTEM                                           2   
HOUSTON METHODIST                                                        0   
HUNTSVILLE HOSPITAL HEALTH SYSTEM                                        0   
IASIS HEALTHCARE LLC                                                     0   
KALISPELL REGIONAL HEALTHCARE                                            0   
LEXINGTON MEDICAL CENTER SYSTEM                                          0   
LIFEPOINT HOSPITALS INC                                                  0   
MEMORIAL HERMANN HEALTHCARE SYSTEM                                       0   
MERCY (FORMERLY SISTERS OF MERCY)                                        1   
MERCY HEALTH MINISTRY SUPPORT SERVICES (FORMERL...                       0   
MERCY HEALTH NETWORK                                                     1   
MIDDLESEX HEALTH SYSTEM                                                  0   
NORTH SHORE-LONG ISLAND JEWISH HEALTH SYSTEM                             0   
NORTHSIDE HOSPITAL HEALTHCARE SYSTEM                                     0   
ROCHESTER REGIONAL HEALTH SYSTEM (AKA VIAHEALTH)                         0   
SAINT TAMMANY PARISH HOSPITAL (SYSTEM)                                   0   
STORMONT-VAIL HEALTHCARE                                                 1   
SUMMA HEALTH SYSTEM                                                      0   
SUTTER HEALTH                                                            0   
TENET HEALTHCARE CORPORATION                                             0   
TEXAS HEALTH RESOURCES                                                   0   
UNIVERSAL HEALTH SERVICES                                                0   
UNIVERSITY HEALTH CARE SYSTEM                                            0   
WELLSTAR HEALTH SYSTEM                                                   0   
WEST VIRGINIA UNITED HEALTH SYSTEM                                       0   

                                                                          \
                                                                           
Region                                             Great Lakes Northeast   
Network/Parent                                                             
"QUEEN'S HEALTH SYSTEMS"                                     0         0   
"SAINT JOSEPH'S CANDLER 